dateUtility.js

```diff
@@ -0,0 +1,107 @@
+/*
+** author: fulin shen
+** description: create a date utility using revealing pattern
+*/
+'use strict';
+
+var dateUtility = (function() {
+  var _privateDate;
+
+  function _initDate() {
+    _privateDate = new Date();
+  }
+
+  function setDate(newDate) {
+    if(newDate != null) {
+      _privateDate = new Date(newDate);
+    } else {
+      _privateDate = new Date();
+    }
+  }
+
+  function getDayName() {
+    var locale = 'en-us';
+    if(!_privateDate) {
+      _initDate();
+    }
+
+    return _privateDate.toLocaleDateString(locale, { weekday: 'long' });
+  }
+
+  function getMonthName() {
+    var locale = 'en-us';
+    if(!_privateDate) {
+      _initDate();
+    }
+
+    return _privateDate.toLocaleString(locale, { month: 'long' });
+  }
+
+  //getDate() and getDate('milliseconds') return milliseconds
+  //getDate('formatted') return date in the form of 'Month Day, Year'
+  function getDate(format) {
```

Comment:
The `format` parameter here should be an object with a format property on it. Usage would look like:

`dateUtility.getDate({ format: "milliseconds" });`